### PR TITLE
feat: add nextcloud compatibility and firebase optionality

### PR DESCRIPTION
**Pull request by zoop-btc**
I would like to propose these changes so it'll be possible to use Nextcloud as a backup mechanism.
This also introduces a new structure to the values. It makes sense to consolidate them as backup since they're both used for the mongodb as well as the scb backup. This is what I came up with but please feel free to give input!
```
backup:
  googlecloud:
    enabled: true
    bucketName: galoy-staging-backups
    secretName: gcs-sa-key
  dropbox:
    enabled: true
    secretName: dropbox-access-token
  nextcloud:
    enabled: false
    secretName: nextcloud-credentials
```

If a user wants to deploy the charts right now without having access to dropbox/google cloud/firebase then it will fail since the secrets referenced don't exist. The only solution I saw was to encapsulate them in if statements and use _firebaseaccountneeded_ as well as the _enabled_ values above. I made it so it'll still reflect the defaults by disabling nextcloud. However disabling firebase will turn off the [daily notifications](https://github.com/GaloyMoney/galoy/blob/main/src/servers/daily-balance-notification.ts), which anybody deciding to forego google needs to be made aware of. Maybe I will tackle that later on.

**Are there any other issues which not using google at all could create?** (I have not noticed any in my test environment.)

This pull request coincides with one [for the galoy repo ](https://github.com/GaloyMoney/galoy/pull/1316).

The secret for nexcloud should look like this and needs to be manually applied like the others:
```
apiVersion: v1
kind: Secret
metadata:
  name: nextcloud-credentials
type: Opaque
data:
  host: <base64encodedfullurl>
  user: <base64encodeduser>
  password: <base64encodedpassword>
```

The host should point to the full url of the nextcloud webdav target like this:
`https://myserver.com/remote.php/dav/files/myuser/backupfolder`

thank you kindly for the consideration
ps: sorry adding so many checks to the mongodb backup script, but better safe than sorry

### CODE REVIEW

**Comment by nicolasburtey** [comment]
thanks for your contribution @zoop-btc !

I didn't dive into the code, but making firebase a non mandatory dependency is very useful

File: charts/galoy/templates/mongo-backup-configmap.yaml
Comment (krtk6160) [inline charts/galoy/templates/mongo-backup-configmap.yaml:25]:
```suggestion

```

File: charts/galoy/templates/mongo-backup-configmap.yaml
Comment (krtk6160) [inline charts/galoy/templates/mongo-backup-configmap.yaml:21]:
```suggestion

```

**Review by krtk6160 — CHANGES_REQUESTED** [review]
The code involving mongo backup tools can be removed as we have now shifted to using a dedicated docker image for the mongo backup cronjob which has the dependency preinstalled

**Comment by zoop-btc** [comment]
Thank you @krtk6160 this looks even better now since the mongodb-tools value can be completely removed :)
I've also cleaned up the pull request, sorry new to github.

**Comment by zoop-btc** [comment]
I have moved the trigger outside of the deployment loop @bodymindarts  as per your suggestion.
This means another change to the values:
https://github.com/GaloyMoney/charts/blob/d2b61bf1212e5668448baf18dab492e33de452b2/charts/galoy/values.yaml#L46-L56

This format would be followed by the other galoy deployments. I can do it for the others as well in a separate PR.
I would also like to make the resources configurable there instead of having them fixed in the template.

**Comment by krtk6160** [comment]
> I have moved the trigger outside of the deployment loop @bodymindarts as per your suggestion. This means another change to the values:
> 
> https://github.com/GaloyMoney/charts/blob/d2b61bf1212e5668448baf18dab492e33de452b2/charts/galoy/values.yaml#L46-L56
> 
> This format would be followed by the other galoy deployments. I can do it for the others as well in a separate PR. I would also like to make the resources configurable there instead of having them [fixed in the template](https://github.com/GaloyMoney/charts/blob/c4cf00c4dc72f8342ff765467f0ca8f275fbb9b8/charts/galoy/templates/deployment.yaml#L65).

@sandipndev iirc you too were refactoring the galoy chart. Does that have any overlap/conflict with the changes here? Might make sense to sync with @zoop-btc to avoid duplicated efforts or conflicts

**Comment by sandipndev** [comment]
@krtk6160 nope, I was doing the secrets so it's not in the scope of this PR at all.

**Comment by zoop-btc** [comment]
updated branch - no breaking changes found - ready for review

File: charts/galoy/values.yaml
Comment (krtk6160) [inline charts/galoy/values.yaml:87]: minor nit: should use camelcase here -> `secretName` to be in line with other key names

Comment (zoop-btc) [reply charts/galoy/values.yaml]: Thank you, is ammended now.

**Comment by krtk6160** [comment]
@zoop-btc There's a significant refactor of the galoy chart that @sandipndev is doing in #1193. It has some amount of overlap with this PR, so those parts will no longer be necessary once we rebase this PR following the merge of #1193.

P.S. No need to rebase now, #1193 is still WIP.

**Comment by zoop-btc** [comment]
@krtk6160 Understood, I'll wait for it to be merged and then adjust this PR accordingly. Thank you for the heads-up.
I'm putting it back to draft mode.

**Comment by zoop-btc** [comment]
closing this, I will split it into two pull requests, one for nextcloud and the other to make firebase optional - I think factoring trigger out of the deployment was too big of a scope

**Comment by sandipndev** [comment]
@zoop-btc the newer refactored charts have been merged. It now allows you to add in another `nextcloud` and `firebase` fields to https://github.com/GaloyMoney/charts/blob/main/charts/galoy/values.yaml#L184 and the overall work would be much cleaner.

**Pull request closed without merging**
